docs/decisions/0013-backend-for-frontend-api.md

```diff
@@ -0,0 +1,69 @@
+of resolving the series of API calls required to resolve the requirements of the feature resulted in the creation of a 
+backend-for-frontend (BFF) layer within the enterprise-access service ([see ADR](https://github.com/openedx/enterprise-access/blob/main/docs/decisions/)).
+
+A BFF layer is a accepted architectural consideration when designing for microservices when the consideration of consolidating
+business logic dependent on multiple services and an intentional reduction of latency between N number of services required to 
+determine the resultant business logic outcome versus a single API call with the logic already resolved to be used.
+
+For the enterprise learner portal implementation, currently, the dashboard route, `/:enterpriseSlug`, is the only route currently
+implementing the BFF layer on _specific_ API calls for the sole intention of supporting, auto apply licenses and auto enrollment via default intentions.
+
+### Making the BFF API call via `useBFF`
+
+At a high level, the implications to the frontend is the inclusion of the following hooks, services, and keys and constraints: 
+
+- `fetchEnterpriseLearnerDashboard` ([source](https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/337a7b44d94d8be5d9233e80e2fa0e2de72d165c/src/components/app/data/services/bffs.ts))
+  - This is the function that makes the service call to the BFF. It requires either a `entepriseSlug` or `enterpriseUuid` to resolve successfully. 
+- Creation of the query key ([source](https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/337a7b44d94d8be5d9233e80e2fa0e2de72d165c/src/components/app/data/queries/queryKeyFactory.js#L267))
+  and helper function `queryEnterpriseLearnerDashboardBFF` ([source](https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/337a7b44d94d8be5d9233e80e2fa0e2de72d165c/src/components/app/data/queries/queries.ts#L270))
+- `useBFF` ([source](https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/337a7b44d94d8be5d9233e80e2fa0e2de72d165c/src/components/app/data/hooks/useBFF.js#L16))
+  - Since the introduction of React Query and custom hooks, the migration process involved abstracting a generalized `useBFF` hook
+    that determines the following conditionals:
+    - Within `resolveBFFQuery` ([source](https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/337a7b44d94d8be5d9233e80e2fa0e2de72d165c/src/components/app/data/queries/utils.js#L11))
+      - Whether a customer is eligible to use the BFF API call by either being explicitly enabled via the configuration API as a defined `enterpriseUuid`,
+        or an individual customer has been selected to resolve the dashboard page with the BFF loader with a gradual rollout process via a Waffle flag.
+      - Whether the current route the user is on is eligible to make the BFF call, if so, return the query required to call the BFF.
```

Comment:
[suggestion] might be worth expanding on what "eligible" means in this case, i.e. the page route has a corresponding BFF API endpoint that exists.